src/Uno.UI.RemoteControl/HotReload/ClientHotReloadProcessor.Xaml.cs

```diff
@@ -137,15 +124,19 @@ private static void PropagateProperties(FrameworkElement oldView, FrameworkEleme
 				return;
 			}
 
-			if (newView.DataContext is null
-				&& oldView.DataContext is not null)
+			var commonProperties = from oldProp in DependencyProperty.GetPropertiesForType(oldView.GetType())
+								   from newProp in DependencyProperty.GetPropertiesForType(newView.GetType())
+								   where oldProp.Name == newProp.Name
+								   let oldValue = oldView.GetValue(oldProp)
+								   let newValue = newView.GetValue(newProp)
+								   where oldValue is not null && newValue is null
+								   select (oldProp, oldValue, newProp);
+			foreach (var p in commonProperties)
 			{
-				// If the DataContext is not provided by the page itself, it may
-				// have been provided by an external actor. Copy the value as is
-				// in the DataContext of the new element.
-
-				newView.DataContext = oldView.DataContext;
+				newView.SetValue(p.newProp, p.oldValue);
```

Comment:
@nickrandolph This is likely to cause trouble with properties value precedence, where GetValue reads from any precedence, but SetValue set only for the local precedence. What's the scenario that is being addressed?

Reply:
ok, so would be better to constraing GetValue to local precedence then? Essentially we want to make sure any local values that would have been set on the xaml element are set on the new instance

Reply:
The intent here is to re-apply local values only, so there's a GetValue with precedence. If the value is not `Unset` then it can be applied to the new instance.

Reply:
@jeromelaban can you confirm [my change](https://github.com/unoplatform/uno/blob/93084456efaa6b7df81f35c63a00029bafe4fcb3/src/Uno.UI.RemoteControl/HotReload/ClientHotReloadProcessor.Xaml.cs#L130) is what makes sense here?

Reply:
@nickrandolph You'll need to add a condition like `oldValue != DependencyProperty.UnsetValue` as well, because getting the local precedence will only return `null` if the `oldValue` was explicitly locally set to `null`. I actually mislead you on the `GetValue` call, reading again the code. [`oldView.ReadLocalValue()`](https://learn.microsoft.com/en-us/uwp/api/windows.ui.xaml.dependencyobject.readlocalvalue?view=winrt-22621#returns) is the right call to only get the local value (regardless of any other precedence).